### ISSUE TITLE
Can we update to latest version 6.5

### ISSUE THREAD

**Issue by Rowleys** [hassio-addons/addon-traccar#334]
Hi. Is there anyway that we can update manually to the lastest version of traccar. 

Thanks

**Comment by LarsNorgaard** [comment]
You can do a fork and edit the dockerfile like this:
https://github.com/hassio-addons/addon-traccar/pull/283/files

**Comment by Rowleys** [comment]
So if I edit the file . 
traccar/Dockerfile
ARG TRACCAR_VERSION="v6.5"

will I need to do an new install or will it update ?

**Comment by LarsNorgaard** [comment]
Ohh I thought it would be as simple as that. I see the docker file already has v6.5 in it, but the server build says it's version 5.12?

**Comment by Rowleys** [comment]
I dont know then. Will the owner not make a update ?

**Comment by ivdimitro** [comment]
@frenck is this add-on still supported? Can we expect update or should we fork it? 
The coming google timeline EoL makes the topic quite relevant ;)

**Comment by petep0p** [comment]
i would also love to see this happen. I know that the Traccar author has an official Docker image (just not specifically for home assistant) here:
https://www.traccar.org/docker/

If i understood how all this works i would do it, but it's currently beyond my ken.

**Comment by martinhoess** [comment]
latest version is now 6.6

https://www.traccar.org/blog/traccar-6-6/
> Notable changes in this release include:
> Managers can now send announcements
> Fixes for multiline SMS notifications
> Early and late computed attributes
> Improvements to user session security
> Support for custom navigation links
> Enhanced speed color scheme
> Collapsible settings and reports drawer
> Improved device card UI
> Event report filtering by alarm type
> Chart report enhancements
> There are no new protocol ports, but we have added support for some new device models and improved support for existing devices.

**Comment by petep0p** [comment]
it looks like there is a pull request in the queue that is asking to update from 6.5 to 6.6...

**Comment by DanielNagy** [comment]
> Ohh I thought it would be as simple as that. I see the docker file already has v6.5 in it, but the server build says it's version 5.12?

Other than v6.6 just being released a few days ago. It needs a new release built.
ie https://github.com/hassio-addons/addon-traccar/releases only shows v0.25.0 done back Feb 2024.

@frenck, the renovate bot appears to correctly updating the docker file to get the later version of traccar v6.6.
however the last action i see ran is this for v.6.5, and it only updated the Edge repo - https://github.com/hassio-addons/addon-traccar/actions/runs/11793092549/job/32847993533

Tried the Edge repo version of v6.5, and it crashes, fails to start. I assume this is why it hasnt proceeded from edge

**Comment by DanielNagy** [comment]
Fyi - this is the addon logs i get from the Edge Traccar, not very helpful debug log.

```
-----------------------------------------------------------
 Add-on: Traccar
 Modern GPS Tracking Platform
-----------------------------------------------------------
 Add-on version: ff8297e
 You are running the latest version of this add-on.
 System: Home Assistant OS 14.1  (amd64 / qemux86-64)
 Home Assistant Core: 2025.1.2
 Home Assistant Supervisor: 2024.12.3
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
Log level is set to DEBUG
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/mysql.sh
[11:15:34] DEBUG: Requested API resource: http://supervisor/services/mysql
[11:15:34] DEBUG: Request method: GET
[11:15:34] DEBUG: Request data: {}
[11:15:34] DEBUG: API HTTP Response code: 200
[11:15:34] DEBUG: API Response: {"result":"ok","data":{"host":"core-mariadb","port":3306,"username":"service","password":"xxx","addon":"core_mariadb"}}
cont-init: info: /etc/cont-init.d/mysql.sh exited 0
cont-init: info: running /etc/cont-init.d/nginx.sh
cont-init: info: /etc/cont-init.d/nginx.sh exited 0
cont-init: info: running /etc/cont-init.d/traccar.sh
cont-init: info: /etc/cont-init.d/traccar.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun nginx (no readiness notification)
services-up: info: copying legacy longrun traccar (no readiness notification)
[11:15:34] INFO: Starting Traccar server...
s6-rc: info: service legacy-services successfully started
[11:15:36] WARNING: Traccar crashed, halting add-on
[11:15:36] INFO: Traccar stopped, restarting...
s6-rc: info: service legacy-services: stopping
[11:15:36] INFO: NGINX stopped, restarting...
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service base-addon-log-level: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service base-addon-log-level successfully stopped
s6-rc: info: service base-addon-banner: stopping
s6-rc: info: service base-addon-banner successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

**Comment by martinhoess** [comment]
not stale

**Comment by petep0p** [comment]
yes, i am also still anxiously awaiting the help of someone much smarter than me....

**Comment by jrb80** [comment]
Signing up for the v6.6 bump!

**Comment by BreitiDE** [comment]
hey guys, have tried to generate my own image based on dockerfile but was not able to get it running as a addon ... starting the image on HA manualy with portainer failes ... would be great if @frenck can have a look into this ... thanks

**Comment by martinhoess** [comment]
not stale

**Comment by sheckandar** [comment]
It is easy to update traccar jar inside the container without building a new add-on for HA.

Download zip file from Traccar home page: https://www.traccar.org/manual-installation/

Copy the file to your HA installation (I have Samba add-on, so I copied the file to `\\<samba ip>\addon_configs\a0d7b954_traccar`

Sign into a SSH session of your HA and then (make sure traccar add-on is running)

```
docker container exec -ti addon_a0d7b954_traccar bash
apk add openjdk17-jre-headless
cd /opt/traccar
unzip -o /config/traccar-other-6.7.1.zip
exit
docker ps -a
```
Get the docker container id for traccar add-on
```
docker commit <container-id> ghcr.io/hassio-addons/traccar/amd64
docker rmi ghcr.io/hassio-addons/traccar/amd64:0.25.0 -f
docker image tag ghcr.io/hassio-addons/traccar/amd64 ghcr.io/hassio-addons/traccar/amd64:0.25.0

```
Restart add-on in HA

**Comment by petep0p** [comment]
@sheckandar i was excited to see this and gave it a try - however, i hit a snag. the only thing i changed was "amd64" to "aarch64" to match all the docker containers on my home assistant green, but the traccar server just continually crashes on start-up (without saying why) after doing all this. would the architecture flag change be causing the crash, do you think? if i entered it all as you wrote it would it work on a home assistant green?

**Comment by sheckandar** [comment]
Those errors are fine. The failure error is 

> Address in use - BindException (...)

Which is also fine since the old version of Traccar was still running. I don't see anything concerning in the log.

Try committing the changes and then restart the add-on. See what happens.

**Comment by petep0p** [comment]
Yeah, it seems like it would start manually if the old instance were down, but i haven't confirmed. However, committing the change to a new image and then restarting the add-on results in the same behavior. Looking back in this thread, it seems like it's the same behavior as trying to run the edge build of the add-on.  So the new version of Traccar works, but there's something about the way the add-on is built that it doesn't play well with. 

@sheckandar what hardware are you running?

**Comment by petep0p** [comment]
also, i just looked at the dockerfile here in the main branch and notice that it's adding a lot more packages. maybe that's what's missing from this overwriting method: 

![Image](https://github.com/user-attachments/assets/882d055f-053a-4aa1-b7c7-9e71cef08442)

once I get a minute to do this I'll give it a try

**Comment by sheckandar** [comment]
> @sheckandar what hardware are you running?

It's just an old PC with an AMD CPU. Nothing special. HA is running inside a VM.

Do you have any custom settings that you set in Traccar ? Are there any xml files in the /config directory inside the container  ?Sometimes a setting is removed in a newer version and can causes issues. If there is one, just rename it to .old temporarily and restart the add-on. If not, another option is nuking all your Traccar data and starting over completely.

But the thing is that without an actual error, all we have are guesses.

**Comment by petep0p** [comment]
yeah, i even set the log level to debug, but it gave no further information about the crash. i do have xml settings, but they're pretty basic (filtering, computed attributes, etc). i will try it with that removed. 

btw, adding/upgrading those packages made no difference - they were already there.

**Comment by petep0p** [comment]
well, it's still a fully functional tracking server at v5.12...

**Comment by petep0p** [comment]
Can anyone else running HA Green try the docker image edit method above and see if it works for them?

**Comment by martinhoess** [comment]
not stale

**Comment by petep0p** [comment]
@frenck is there any info you can give us to get this unblocked? we are trying to solve it on our own but the problem seems to be some subtle aspect of HAOS-to-addon interaction that we don't yet understand well enough to get behind. it seems like most of the system is set up to automatically keep it up to date, but there is just one setting somewhere that changed between traccar v5 and v6 is that is keeping the add-on from working.

**Comment by petep0p** [comment]
not stale

**Comment by petep0p** [comment]
@danielsmith89 does the traccar integration still work with your docker server?

**Comment by petep0p** [comment]
also, has anyone taken a stab at just creating a new custom addon? i haven't yet learned how to do that

**Comment by danielsmith89** [comment]
> also, has anyone taken a stab at just creating a new custom addon? i haven't yet learned how to do that

That's what I tried last night. For the life of me I couldn't get Traccar to start successfully. That's why I've now resorted to running in Docker separately to HA instead.

**Comment by danielsmith89** [comment]
> ok got it. i thought you were referring to the trick earlier in the thread of modifying the docker image of the existing add-on

Both achieve effectively the same result (it doesn't work!).

**Comment by DanielNagy** [comment]
> [@danielsmith89](https://github.com/danielsmith89) out of curiosity, does the docker server need it's own database, or is it possible to have it record to the HA database?

My suggestion is to keep it a separate database. Traccar will consume alot of space over time. You wouldn't want that effecting your Home Assistant.

**Comment by danielsmith89** [comment]
> @danielsmith89 out of curiosity, does the docker server need it's own database, or is it possible to have it record to the HA database?

My understanding is that the HA addon database can only be accessed by other addons/HA itself and not to the "outside world" so you'll need to run another db. 

There is a Docker Compose example on the Traccar website for setting up both Traccar and the db for it.

**Comment by petep0p** [comment]
I wanted to report that I've also given up on this add-on, as much as I've loved it. I wanted to share my solution:
background:

- I have my home network set up w/ tailscale, and HA is set up with funnel to be accessible from outside the network.
- i had an Rpi4B laying around that i didn't know what to do with

the way around it all:

- i bought an m.2 NVMe base for the Rpi4B, and a POE hat
- I installed 64-bit RpiOS onto it
- I installed tailscale, NGINX (via docker) (perhaps not strictly necessary, but i did it), mariadb, and the latest traccar server
- set up funnel in tailscale to the Rpi, running through NGINX reverse proxy and mapping port 443, 8443, or 10000 to port 8082 for the traccar web application (since i've already gone through this w/ the HA instance)
- set up DDNS on my router and port forwarded the GPS devices' port to the Rpi.
- set my GPS devices to send to the router's DDNS addy
- (here's where i got hung up for a bit) opened the firewall on the Rpi for the GPS device port
- added the instance in the HA traccar server integration (here's where i got hung up again: b/c of all the VPN/funnel/reverse proxy'ing, i forgot that the port to specify in the integration setup screen is 443, 8443, or 10000 (the only allowed tailscale funnel ports), not the default 8082)

And now I've got my self-hosted GPS tracking server running and integrated into HA, completely updateable and self-manageable.
does an Rpi count as bare metal?

UPDATE: i wanted to add that i got rid of the NGINX reverse proxy step b/c it was actually causing issues w/ the tailscale certs. done that way, tailscale will not auto-manage the certs, and they will end up expiring. the right way is to just run the funnel command like:
`sudo tailscale funnel --bg --https={PORT} localhost:8082`
where `{PORT} = 443/8443/10000` depending on your setup. Also, you can change the localhost port to whatever you like, so long as the server is set to listen there.

**Comment by petep0p** [comment]
historical data wasn't important to me, so i just started fresh. (if you know how to work with MariaDB, i imagine it could be as simple as exporting tables from the HA database and importing them into the new one, but don't quote me on that)

as far as the server, user, and device settings i just manually copied things into the new setup. there is enough different with the traccar.xml and the HA default one it inherited from that i didn't want to muck things up trying to copy and paste. 

i only have 1 admin tracking 2 devices so that wasn't a big deal for me.

**Comment by petep0p** [comment]
come to think of it the configuration in the add-on container could very well be the issue we were having.... that is one of the things @sheckandar mentioned that i never got around to debugging

**Comment by martinhoess** [comment]
@petep0p Ah ok! I have “lost” all the data when switching from the old DB to MariaDB so this time I’d really like to keep everything. But with the DB it should hopefully be enough to just copy it. I’ll probably need to check the settings though, because if I remember correctly I had some extra configurations for protocols.

**Comment by MrFinchMkV** [comment]
Hey,

does anyone know why the current main doesn't get released?
I saw roughly a month ago that changes updating Traccar to version 6.10 were merged, but no release until now :/

Kind regards :)

**Comment by petep0p** [comment]
> Hey,
> 
> does anyone know why the current main doesn't get released? I saw roughly a month ago that changes updating Traccar to version 6.10 were merged, but no release until now :/
> 
> Kind regards :)

there is something in the configuration of the add-on that causes it to crash with anything above v5.12. there is currently no maintainer to rectify the issue. it's far less work just to set up a current version server on another piece of hardware (see above in this thread) and link it to the core integration, which still works with an external server. that is, unless you're familiar enough with both HA addon config, traccar config, and their interaction to debug it on your own?

**Comment by MrFinchMkV** [comment]
Thanks for the explanation.
I sadly don't see me having time to debug that anytime soon.
If I unexpectedly find the time and discover something, I will post it.

**Comment by MrFinchMkV** [comment]
Happy new year everyone! :)

I got some time to setup a dev environment and test a littlebit.
What I figured out:
- That last starting version is 6.1
- Traccar is crashing before it starts to log ANYTHING
- Updating openjdk from version 17 to 21 didn't help

Would be nice if 1 or 2 others could confirm that Traccar v6.1 is working.

Maybe would could than at least get an update to this version.

I'm now trying to find a way to figure out why v6.2 is crashing.

**Comment by MrFinchMkV** [comment]
Hey again :P

As I thought:
The problem is the configuration.
In version 6.2 'config.default' was removed.
- https://www.traccar.org/blog/traccar-6-2/
- https://github.com/traccar/traccar/compare/v6.1...v6.2

Currently I have no idea how to migrate or combine the current default and the current custome config to a unified one so existing installations don't brake.

**Comment by petep0p** [comment]
I'm not sure if you've already read through these: 

https://www.traccar.org/configuration-file/
https://www.traccar.org/forums/topic/62-release-default-configuration-file-is-now-removed-to-avoid-confusion-and-misconfiguration/page/2/

from what i understand, any change to the default (as specified in the documentation link on each key's definition) is just spelled out in the custom config file. disabling a key would be setting it to an empty value.

is that anything like what you're asking?

**Comment by MrFinchMkV** [comment]
Take a look at the merge request.
I decided to create a backup of the current configuration and create a new one.
Ps. Yeah I read the documentation and the forum post :)

**Comment by petep0p** [comment]
> Take a look at the merge request. 

it looks like the request is missing a required label (bug/breaking change/etc). you might need to re-submit it with the label added during the original submission - i seem to remember not being able to edit the label unless you're a code owner. i could be wrong, though.

**Comment by MrFinchMkV** [comment]
After not finding the option to add a label I thought the same. That the codeowner has to add the missing information, like labels.

**Comment by petep0p** [comment]
that doesn't bode well for the pull request...

**Comment by jrb80** [comment]
@MrFinchMkV have you considered forking as it's clear the current code owner is not interested in maintaining it?

**Comment by DAVIZINH0** [comment]
maybe the owner can help with this? @frenck 
this is a very usefull addon, and a so older version maybe will be a problem of security

**Comment by MrFinchMkV** [comment]
> [@MrFinchMkV](https://github.com/MrFinchMkV) have you considered forking as it's clear the current code owner is not interested in maintaining it?

I have. But as this is part of the official community addons I hoped there is more activity.
In December I said I don't have time to analyse the problem.
A few days/weeks later I suddenly had.

Maybe you wake up one day and either @frenck commented or merged OR I forked it.
Currently I'm trying not to come in the position to be responsible :D

**Comment by JohnWallerUK** [comment]
Can someone please update to the latest version?

**Comment by MrFinchMkV** [comment]
@JohnWallerUK, @frenck is the only who can do it.

**Comment by MrFinchMkV** [comment]
Does anyone already have a forum account?
Is it maybe possible to get attention over there?

I get that this is not high priority repository but regular maintenance should be done.
I mean after all it's still part of the official community apps.

**Comment by petep0p** [comment]
@MrFinchMkV unfortunately, you get the same level of attention in the forum. i don't think anything in the "community" app category is "official" at all, despite the naming. the only way to get timely updates on most of these projects is to fork them.

**Comment by MrFinchMkV** [comment]
I don't want to promise anything.
But because I don't want to host traccar on a different Server (what, at least for my, would be much easier :D) I will try to figure out how to get the existing pipelines to work on my fork.
But summer is around the corner, so please don't expect too much to happen in the near future.
